helpers/lib/resourceHints.js

```diff
@@ -0,0 +1,64 @@
+/* eslint-disable curly */
+const _ = require("lodash");
+const utils = require("handlebars-utils");
+
+const resourceHintsLimit = 50;
+
+const defaultResourceHintSate = 'preload';
+
+const resourceHintStates = [defaultResourceHintSate, 'preconnect', 'prefetch'];
+
+const resourceHintFontType = 'font';
+const resourceHintStyleType = 'style';
+const resourceHintScriptType = 'script';
+const resourceHintAllowedTypes = [resourceHintStyleType, resourceHintFontType, resourceHintScriptType];
+
+/**
+ * @param {string} path - The uri to the resource.
+ * @param {string} state - 'preload' or 'preconnect'
+ * @param {string} type - any of [style, font, script]
+ */
+function addResourceHint(globals, path, state, type) {
+
+    path = _.trim(path);
```

Comment:
I would also recommend you to check if the `path` is a string, otherwise throw an error with some distinct message